### ISSUE TITLE
pygame.mixer.Sound.get_num_channels() dosen't work as intended

### ISSUE THREAD

**Issue by galexandreg** [pygame/pygame#1430]
I tried to look at the SDL mixer docs and the mixer.c source to understand the problem, but I am a complete beginner with all that stuff, so I may be completly wrong.

According to the documentation, get_num_channels will return the number of active channels the sound is playing on. But it seems that if I play a sound and let it finish, the number will stay up until another sound uses the same channel.

Add any sound to line 12 and 13 to test...
```
import sys
import pygame

pygame.init()
clock = pygame.time.Clock()

# My mac dosen't like when I press keys without a window.
window = pygame.display.set_mode((100, 100))
window.fill((0, 0, 0))
pygame.display.flip()

sound_1 = pygame.mixer.Sound("testsound")
sound_2 = pygame.mixer.Sound("testsound")

while True:
    for e in pygame.event.get():
        if e.type == pygame.QUIT:
            pygame.quit()
            sys.exit()
        if e.type == pygame.KEYUP:
            if e.key == pygame.K_ESCAPE:
                pygame.quit()
                sys.exit()
            if e.key == pygame.K_1:
                sound_1.play()
            elif e.key == pygame.K_2:
                sound_2.play()

    print("sound_1.get_num_channels: " + str(sound_1.get_num_channels()))
    clock.tick(10)
```

Pressing 1 plays the first sound and 2 the second. The terminal will print the result of sound_1.get_num_channels().

If I play the sound_1 once, the number will always be 1 until I play the second sound and the same channel is used. If I play the first sound multiple times before it's done, it will use multiple channels and they will always show in get_num_channels until I play something else that will use the same channels.

It seems that in order to do that, the channel is added to an SDL_mixer group specific to that sound on line 638 of mixer.c:
`Mix_GroupChannel(channelnum, (intptr_t)chunk);`

And when get_num_channels() is called on a mixer.Sound, it will find the channels on the same "sound group"
mixer.c line 644
```
static PyObject *
snd_get_num_channels(PyObject *self, PyObject *args)
{
    Mix_Chunk *chunk = pgSound_AsChunk(self);
    MIXER_INIT_CHECK();
    return PyInt_FromLong(Mix_GroupCount((intptr_t)chunk));
}
```

But the channel is kept on that group, even when the sound is not playing anymore.

I may or may not try a pull request later if it still makes sense, but I never really used C much and never used the SDL directly, so the fix might be wrong...

**Comment by illume** [comment]
Hi,

I could reproduce it. Changed your code a little to use the sounds included in the pygame.examples.

```python
import sys
import os
import pygame

pygame.init()
clock = pygame.time.Clock()

# My mac dosen't like when I press keys without a window.
window = pygame.display.set_mode((100, 100))
window.fill((0, 0, 0))
pygame.display.flip()

import pygame.examples
main_dir = os.path.split(os.path.abspath(pygame.examples.__file__))[0]
sound_1 = pygame.mixer.Sound(os.path.join(main_dir, "data", "car_door.wav"))
sound_2 = pygame.mixer.Sound(os.path.join(main_dir, "data", "secosmic_lo.wav"))


while True:
    for e in pygame.event.get():
        if e.type == pygame.QUIT:
            pygame.quit()
            sys.exit()
        if e.type == pygame.KEYUP:
            if e.key == pygame.K_ESCAPE:
                pygame.quit()
                sys.exit()
            if e.key == pygame.K_1:
                sound_1.play()
            elif e.key == pygame.K_2:
                sound_2.play()

    print("sound_1.get_num_channels: " + str(sound_1.get_num_channels()))
    clock.tick(10)
```

**Comment by galexandreg** [comment]
Fixed in #1431 😄 Thanks!
